### ISSUE TITLE
Thread pools in StatsDNonBlockingProcessor not properly closed

### ISSUE THREAD

**Issue by domisum** [DataDog/java-dogstatsd-client#116]
My application using this library doesn't properly shut down. I already updated to the latest version 2.10.2 which fixed another thread pool issue (PR #115 ) , but this one seems to be still wrong.

```
Thread[pool-3-thread-1,5,main], id: 19, daemon: false
    java.base@11.0.7/java.lang.Thread.sleep(Native Method)
    app//com.timgroup.statsd.StatsDNonBlockingProcessor$ProcessingTask.run(StatsDNonBlockingProcessor.java:35)
    java.base@11.0.7/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
    java.base@11.0.7/java.util.concurrent.FutureTask.run(FutureTask.java:264)
    java.base@11.0.7/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    java.base@11.0.7/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    java.base@11.0.7/java.lang.Thread.run(Thread.java:834)
```

**Comment by blevz** [comment]
I ran into a similar issue when upgrading my dogstatsd client.  I believe the issue is the StatsDNonBlockingProcessor executor is using non daemon threads.  Additionally, if you opt into using telemetry, the TimerTask it spawns is non daemon and will similarly hang the JVM at shutdown.  I have a pull request open with the changes which should fix both of these issues:
https://github.com/DataDog/java-dogstatsd-client/pull/117

**Comment by domisum** [comment]
Fixed in PR #117
